libsolidity/interface/CompilerStack.cpp

```diff
@@ -1245,6 +1246,27 @@ void CompilerStack::storeContractDefinitions()
 			}
 }
 
+void CompilerStack::annotateInternalFunctionIDs()
+{
+	for (Source const* source: m_sourceOrder)
+	{
+		if (!source->ast)
+			continue;
+
+		for (ContractDefinition const* contract: ASTNode::filteredNodes<ContractDefinition>(source->ast->nodes()))
+		{
+			ContractDefinitionAnnotation& annotation =
+				m_contracts.at(contract->fullyQualifiedName()).contract->annotation();
+
+			for (auto function: FunctionCallGraphBuilder::internalDispatchFunctions(**annotation.creationCallGraph))
+				function->annotation().internalFunctionID = function->id();
+
+			for (auto function: FunctionCallGraphBuilder::internalDispatchFunctions(**annotation.deployedCallGraph))
+				function->annotation().internalFunctionID = function->id();
```

Comment:
Ah, you're currently assigning AST IDs as internal function IDs here? That's actually interesting, but looking forward to potentially having jump tables, we should keep generating tightly packed IDs.
Also the set of functions in the dispatch of the deployed call graph should include all of the ones in the creation call graph, shouldn't it? We also need a deterministic order for generating the IDs here - the edges in the control graph are already sorted by AST ID, so maybe let's just skip the entire ``FunctionCallGraphBuilder::internalDispatchFunctions`` thing and do something like the following here:

```suggestion
uint64_t internalFunctionID = 0;
if (auto const* deployTimeInternalDispatch = util::valueOrNullptr((*annotation.deployedCallGraph)->edges, CallGraph::SpecialNode::InternalDispatch))
	for (auto const& node: *deployTimeInternalDispatch)
		if (auto const* callable = get_if<CallableDeclaration const*>(node))
			if (auto const* function = dynamic_cast<FunctionDefinition const*>(callable))
			{
				solAssert(!function->annotation().internalFunctionID.has_value()); // alternatively, and maybe preferably, we could make it a SetOnce<optional<uint64_t>> in the annotation and skip the assertion here
				function->annotation().internalFunctionID = internalFunctionID++;
			}
if (auto const* creationTimeInternalDispatch = util::valueOrNullptr((*annotation.creationCallGraph)->edges, CallGraph::SpecialNode::InternalDispatch))
	for (auto const& node: *creationTimeInternalDispatch)
		if (auto const* callable = get_if<CallableDeclaration const*>(node))
			if (auto const* function = dynamic_cast<FunctionDefinition const*>(callable))
				// Make sure the function already got an ID since it also occurs in the deploy-time internal dispatch.
				solAssert(function->annotation().internalFunctionID.has_value());
```

Haven't tested this, but something like that should be enough to just get us deterministic tightly packed internal function IDs for all functions.

Reply:
> Ah, you're currently assigning AST IDs as internal function IDs here?

Yes, I thought that this was what I was supposed to do haha

> Also the set of functions in the dispatch of the deployed call graph should include all of the ones in the creation call graph, shouldn't it?

Right, I just remembered now that you mentioned that. I will do the changes you suggested.